### PR TITLE
Add CKE multi-host test setup tools

### PR DESCRIPTION
**Pull request by dulltz**
Related: https://github.com/cybozu-go/cke/pull/37

Add following tools written by Go

- `create-disk-if-not-exists`
  To reduce mtest executing time,
  prepare qcow image contains `/var/lib/docker`.
  if there is no qcow file on cloud storage `cke-test` bucket,
  create GCE instance, then execute create-qcow on it.

- `create-qcow`
  Crepare qcow image, then upload it on `cke-bucket`.

Add CI job to test go code

### CODE REVIEW

File: neco-test/cke/cli/cli.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	AuthProviderX509CertURL string `json:"auth_provider_x509_cert_url"`
+	ClientX509CertURL       string `json:"client_x509_cert_url"`
+}
+
+// LoadAccountJSON returns ServiceAccount from its key file JSON
+func LoadAccountJSON(filename string) (*ServiceAccount, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
```
Comment (morimoto-cybozu) [inline neco-test/cke/cli/cli.go]: `defer file.Close()`

File: neco-test/cke/create-qcow/main.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	"cloud.google.com/go/storage"
+	"github.com/cybozu-go/cke"
+	"github.com/cybozu-go/log"
+	"github.com/cybozu/neco-gcp/neco-test/cke/cli"
+)
+
+var (
+	flgZone        = flag.String("zone", "asia-northeast1-c", "GCP compute zone name")
+	flgBucket      = flag.String("bucket", "cke-test", "GCP Cloud Storage's bucket name")
+	flgAccountJSON = flag.String("account-json", "", "file path of service account's account-dummy.json ")
```
Comment (morimoto-cybozu) [inline neco-test/cke/create-qcow/main.go]: dummy ではなさそう。

File: neco-test/cke/create-disk-if-not-exists/main.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+		Zone:         *flgZone,
+		InstanceName: *flgInstance,
+	}
+
+	targetFile := fmt.Sprintf("var-lib-docker.%x.img.bz2", checkSum)
+	if client.existsTarget(ctxWithTimeout, targetFile) {
+		log.Info(targetFile+" is already exists", nil)
+		os.Exit(0)
+	}
+	log.Info(targetFile+" does not exist, I will start create it", nil)
```
Comment (mitsutaka) [inline neco-test/cke/create-disk-if-not-exists/main.go]: I will start create it -> I will start to create it

File: neco-test/cke/create-disk-if-not-exists/main.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	if err != nil {
+		log.ErrorExit(err)
+	}
+
+	err = client.gcloud("compute", "scp", "--project", client.Project, "--zone", client.Zone, *flgAccountJSON, "./create-qcow", "cybozu@"+*flgInstance+":")
+	if err != nil {
+		log.ErrorExit(err)
+	}
+
+	err = client.ssh("sudo /home/cybozu/create-qcow -account-json=/home/cybozu/account-dummy.json")
```
Comment (mitsutaka) [inline neco-test/cke/create-disk-if-not-exists/main.go]: ```"sudo /home/cybozu/create-qcow -account-json=/home/cybozu/"+*flgAccountJSON ```

File: neco-test/cke/create-disk-if-not-exists/main.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+	if err != nil {
+		log.ErrorExit(err)
+	}
+
+	err = c.gcloud("compute", "scp", "--project", c.Project, "--zone", c.Zone, *flgAccountJSON, "./create-qcow", "cybozu@"+*flgInstance+":")
+	if err != nil {
+		log.ErrorExit(err)
+	}
+
+	err = c.ssh("sudo /home/cybozu/create-qcow -account-json=/home/cybozu/" + *flgAccountJSON)
```
Comment (morimoto-cybozu) [inline neco-test/cke/create-disk-if-not-exists/main.go]: `-account-json=./files/account.json` のような場合、コピー先にディレクトリ構造は作られず
`/home/cybozu/account.json` ができるので、 `path.Base(*flgAccountJSON)` で。

File: neco-test/cke/create-disk-if-not-exists/main.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func (c client) existsTarget(ctx context.Context, target string) bool {
+	_, err := c.Bucket(*flgBucket).Object(target).Attrs(ctx)
+	if err == storage.ErrObjectNotExist {
+		return false
+	} else if err != nil && err != storage.ErrObjectNotExist {
```
Comment (morimoto-cybozu) [inline neco-test/cke/create-disk-if-not-exists/main.go]: 後半が冗長か。

**Pull request merged**